### ISSUE TITLE
compiler warning for time arrays

### ISSUE THREAD

**Issue by timn** [ros/gencpp#37]
Consider the following message spec:
```
time[] foo
```
The C++ code generated for this message generates a warning which concerns me, as it indicates invalid code (the actual warning is much longer, cf. instructions below to reproduce):
```
/usr/lib64/ros/include/ros/serialization.h:451:13: warning: ‘void* memcpy(void*, const void*, size_t)’
writing to an object of non-trivially copyable type ‘__gnu_cxx::__alloc_traits<std::allocator<ros::Time>,
ros::Time>::value_type’ {aka ‘class ros::Time’}; use copy-assignment or copy-initialization instead [-Wclass-memaccess]
       memcpy(&v.front(), stream.advance(data_len), data_len);
```
It seems that some type trait is missing for such time arrays, such that the proper VectorSerializer is chosen.

You can find an example repository at https://github.com/timn/ros-timebug_msgs. I have tried this with ROS Kinetic. Once it is built in a workspace it prints the warning.

**Comment by dirk-thomas** [comment]
Please consider to provide a pull request for this.

**Comment by dirk-thomas** [comment]
Addressed by ros/roscpp_core#102.
